main.cc

```diff
@@ -62,9 +62,23 @@ int main(int argc, char *argv[]) {
     while (!input.eof()) {
         line_no++;
         getline(input, line);
+
+	// Strip any comments that start with ; to the end of the line
+	size_t comment_start = line.find(";");
```

Comment:
Can you make this `line.find_first_of(';')` ? That way, the string implementation can use a cheaper character finding way instead of doing a more expensive string-search.

The result is not modified later, so for readability, I'd suggest to use const
```
  const size_t comment_start = line.find_first_of(';');
```